### ISSUE TITLE
Ubuntu 14.04, OpenSSL, Python 2.7.6 and requests

### ISSUE THREAD

**Issue by daaray** [caktus/django-project-template#179]
Copying and pasting some notes from pressweb project.  

*However, another round of issues has cropped up and we are working through them. We provision on Ubuntu 14.04, which by default comes with python 2.7.6. Now that we are using SSL to serve the site, a combination of OpenSSL, requests, and boto is causing an SSLError on downloading the packaged zipfiles of media:*

*SSLError: [Errno 1] _ssl.c:510: error:14077410:SSL
 routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure*

*I went back to staging and replicated the issue there as well. Various resources [(example)](https://github.com/coursera-dl/coursera/issues/330) indicate that upgrading the system python to 2.7.9 is an appropriate solution to this issue.* 

`conf/salt/project.venv.sls` changes

```
deadsnakes-python2.7:
  pkgrepo.managed:
    - humanname: Deadsnakes PPA for 2.7
    - ppa: fkrull/deadsnakes-python2.7

python-pkgs:
  pkg:
    - installed
    - names:
      - python{{ pillar['python_version'] }}
      - python{{ pillar['python_version'] }}-dev
    - require:
      - pkgrepo: deadsnakes-python2.7
```

TL;DR: If we have projects on Python2.x, we likely need a path to upgrade to 2.7.9

**Comment by dpoirier** [comment]
Don't forget that requirements for `pkgrepo` have to be in
the pkgrepo state using require_in:

    xxx:
        pkgrepo.managed:
            - require_in:
               - pkg: that_needs_pkgrepo

**Comment by mlavin** [comment]
Upgrading to Python 2.7.9 from 2.7.X would require a full rebuild of the virtualenv correct?

**Comment by daaray** [comment]
Yeah, it does. Unsure if there is a way to automate that. I manually blew
away the venv on the server and then did another `fab env deploy` to ensure
it was working as expected.

**Comment by dpoirier** [comment]
Maybe run "venv/bin/python -V" and see if the reported version matches {{ python_version }} and if not, we need to rebuild the venv?

**Comment by dpoirier** [comment]
BTW with current Margarita you can just include the `python.27` state and it'll install the latest Python 2.7 from deadsnakes for you.

You'll still need a way to rebuild the venv, though.

**Comment by vkurup** [comment]
Tracked by caktus/margarita#64 since the states have been moved there.
